client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -442,17 +460,28 @@ export function* copyFile(src, dest, index) {
       type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
       payload: { status: 'ERROR', key: index, operation: 'copy' }
     });
+    return 'ERR';
   }
+  return 'SUCCESS';
 }
 export function* copyFiles(action) {
   const { dest } = action.payload;
   const copyCalls = action.payload.src.map(file => {
     return call(copyFile, file, dest, file.id);
   });
-  yield race({
+  const { result } = yield race({
     result: all(copyCalls),
     cancel: take('DATA_FILES_MODAL_CLOSE')
   });
+  if (!result.includes('ERR'))
+    yield put({
+      type: 'ADD_TOAST',
+      payload: {
+        message: `${
+          result.length > 1 ? `${result.length} files` : 'File'
+        } copied to ${action.payload.dest.name}`
+      }
+    });
```

Comment:
Should we add a unit test to cover these batch cases?

Reply:
I'll add cases for the existing unit tests, but we should make a new task for sagas that are currently untested since unit testing the move/rename/download sagas would quadruple the scope of work for this PR.